src/mapviewAPI.ts

```diff
@@ -0,0 +1,13 @@
+import { App } from 'obsidian';
+import { GeoSearcher, GeoSearcherProvider } from './geosearch';
+import { PluginSettings } from 'src/settings';
+
+export class MapviewAPI {
+    public searchProvider: GeoSearcherProvider;
```

Comment:
Sorry about the misunderstanding; that's not what I meant in our discussion.
If you expose the `searchProvider` and `searcher` fields as an API this way, then you actually expose `geosearch.OpenStreetMapProvider`, `geosearch.GoogleProvider`, `GooglePlacesAPI` and `GeoSearcher`, and all of then now become an API that must be maintained. Every single thing we ever change in `GeoSearcher` is a potential API breakage and that's a lot of maintenance burden to consider.
What I think we should do instead is that this `MapViewAPI` class exposes a single method `geosearch` with the least dependencies possible, or maybe 2-3 methods if you want to provide some more functionality, and mask away all of the implementation.

Reply:
The main reason I want to expose those API calls directly is that abstracting them away again makes it hard to manipulate the raw data (which in my case I want to do, understand if this isn't the direction you want to take it).

How would you suggest exposing the raw response from the search API and things like the Google Places API details endpoints?

Other then exposing the searchProvider directly I don't see a nice way.

If those APIs change would the release not just be a breaking change.